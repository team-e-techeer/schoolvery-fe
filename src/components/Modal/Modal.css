* {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
}
html,
body {
  min-height: 100%;
  height: 100%;
  font-family: helvetica neue, helvetica, arial, sans-serif;
}

.content {
  min-height: 100%;
  height: 100%;
  background: #fff;
  position: relative;
  z-index: 0;
}
.content h1 {
  /* padding: 75px 0 30px 0; */
  text-align: center;
  font-size: 30px;
  line-height: 30px;
}

/* modal style */
html.modal-active,
body.modal-active {
  overflow: hidden;
}
#modal-container {
  position: fixed;
  left: 0;
  top: 0;
  display: table;
  height: 100%;
  width: 100%;
  z-index: 1;
  /* hidden 처리 */
  transform: scale(0);
}
#modal-container .modal-background {
  display: table-cell;
  background: rgba(0, 0, 0, 0.8);
  text-align: center;
  vertical-align: middle;
}
#modal-container .modal-background .modal {
  background: #fff;
  width: 90%;
  padding: 3rem 1rem;
  box-shadow: 0 0 6px 0 rgba(0, 0, 0, 0.5);
  display: inline-block;
  border-radius: 0.7rem;
  position: relative;
}
#modal-container .modal-background .modal h2 {
  font-size: 25px;
  line-height: 25px;
  margin-bottom: 15px;
}
#modal-container .modal-background .modal p {
  font-size: 18px;
  line-height: 22px;
}

/*
 * ------------------------------
 * Unfolding Animation
 * ------------------------------
 */
#modal-container.one {
  /*transform: scaleX(0) scaleY(1);*/
  animation: unfoldIn 0.5s cubic-bezier(0.165, 0.84, 0.44, 1) forwards;
}
#modal-container.one .modal-background .modal {
  /* 초기 display: none 처럼 크기를 0 으로 하여 보여주지 않음*/
  transform: scale(0);
  /* 0.5 초 동안 애니메이션을 실행, 단 0.8초 지연시켜 0.8초 후에 애니메이션을 실행 */
  animation: zoomIn 0.5s cubic-bezier(0.165, 0.84, 0.44, 1) forwards;
}
#modal-container.one.out {
  transform: scale(1);
  animation: unfoldOut 0.5s 0.2s cubic-bezier(0.165, 0.84, 0.44, 1) forwards;
}
#modal-container.one.out .modal-background .modal {
  animation: zoomOut 0.5s cubic-bezier(0.165, 0.84, 0.44, 1) forwards;
}
/* Dimmed showing */
@keyframes unfoldIn {
  0% {
    transform: scaleX(0) scaleY(0.005);
  }
  50% {
    /* 0.5 초간 세로 크기는 작은상태로 가로 너비만 확대  */
    transform: scaleX(1) scaleY(1);
  }
  100% {
    /* 0.5 초간 세로 크기를 확대 */
    transform: scaleY(1) scaleX(1);
  }
}
/* Show modal content  */
@keyframes zoomIn {
  from {
    transform: scale(0);
  }
  to {
    transform: scale(1);
  }
}
/* Hide modal content */
@keyframes zoomOut {
  from {
    transform: scale(1);
  }
  to {
    transform: scale(0);
  }
}
/* Dimmed hiding */
@keyframes unfoldOut {
  0% {
    transform: scale(1);
  }
  50% {
    transform: scaleX(1) scaleY(0.005);
  }
  100% {
    transform: scaleX(0) scaleY(0.001);
  }
}
