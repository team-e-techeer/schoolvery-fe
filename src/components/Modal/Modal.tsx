import { css } from '@emotion/react';
import { useCallback, useEffect } from 'react';
import './Modal.css';
import { AiOutlineClose as CloseIcon } from 'react-icons/ai';

interface Props {
  visible: boolean;
  children: JSX.Element;
  setModalVisible: React.Dispatch<React.SetStateAction<boolean>>;
}

export default function Modal({ visible, setModalVisible, children }: Props) {
  useEffect(() => {
    visible && onClickModalVisible();
    return () => {};
  }, [visible]);

  const onClickModalVisible = useCallback(() => {
    const $modalContainer = document.querySelector('#modal-container') as HTMLElement;
    $modalContainer.removeAttribute('class');
    $modalContainer.classList.add('one');
    setModalVisible(false);
  }, []);

  const onClickOther = useCallback(() => {
    const modalContainer = document.querySelector('#modal-container') as HTMLElement;
    modalContainer.classList.add('out');
  }, []);

  return (
    <>
      <div id="modal-container">
        <div onClick={onClickOther} className="modal-background">
          <div className="modal">
            <button
              onClick={onClickOther}
              css={css`
                display: flex;
                width: 100%;
                justify-content: right;
              `}
            >
              <CloseIcon size={25} />
            </button>
            {children}
          </div>
        </div>
      </div>
    </>
  );
}
