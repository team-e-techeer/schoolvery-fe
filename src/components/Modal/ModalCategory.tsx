import colors from '@/constants/colors';
import { css } from '@emotion/react';
import { useState } from 'react';
import Modal from './Modal';
import Ch from '../../assets/img/ch.png';
import chicken from '../../assets/img/chicken.png';
import Jp from '../../assets/img/jp.png';
import Ko from '../../assets/img/ko.png';
import Pizza from '../../assets/img/pizza.png';
import All from '../../assets/img/all.png';
import Asian from '../../assets/img/asian.png';
import Hamburger from '../../assets/img/hamburger.png';
import koreanFood from '../../assets/img/koreanFood.png';
import coffee from '../../assets/img/coffee.png';
import { useNavigate } from 'react-router-dom';
interface Props {
  visible: boolean;
  setModalVisible: React.Dispatch<React.SetStateAction<boolean>>;
  isNavigating: boolean;
}

interface Info {
  path: string;
  src: string;
  name: string;
  select: boolean;
  id: number;
}

export default function ModalCategory({ visible, setModalVisible, isNavigating }: Props) {
  const navigate = useNavigate();
  const [imageInfo, setImage] = useState<Info[]>([
    { path: '/category/전체', src: All, name: '전체', select: true, id: 0 },
    { path: '/category/한식', src: Ko, name: '한식', select: false, id: 1 },
    { path: '/category/중식', src: Ch, name: '중식', select: false, id: 2 },
    { path: '/category/일식', src: Jp, name: '일식', select: false, id: 3 },
    { path: '/category/아시안', src: Asian, name: '아시안', select: false, id: 4 },
    { path: '/category/치킨', src: chicken, name: '치킨', select: false, id: 5 },
    { path: '/category/피자', src: Pizza, name: '피자', select: false, id: 6 },
    { path: '/category/햄버거', src: Hamburger, name: '햄버거', select: false, id: 7 },
    { path: '/category/분식', src: koreanFood, name: '분식', select: false, id: 8 },
    { path: '/category/커피', src: coffee, name: '커피', select: false, id: 9 },
  ]);
  const onClickCategory = (image: Info) => {
    setImage(imageInfo.map(img => (img.id === image.id ? { ...img, select: true } : { ...img, select: false })));
    isNavigating && navigate(`${image.path}`);
  };
  return (
    <>
      <Modal visible={visible} setModalVisible={setModalVisible}>
        <>
          <span
            css={css`
              display: flex;
              font-size: 2rem;
              margin: 1rem;
              margin-left: 2%;
            `}
          >
            카테고리를 선택해주세요
          </span>
          <div
            css={css`
              display: flex;
              flex-wrap: wrap;
              box-sizing: border-box;
              justify-content: center;
            `}
          >
            {imageInfo.map(image => (
              <button
                onClick={() => onClickCategory(image)}
                css={css`
                  display: flex;
                  flex-wrap: wrap;
                  border-radius: 0.7rem;
                  box-sizing: border-box;
                  background-color: ${image.select ? colors.mainColor : colors.grey100};
                  padding: 1rem;
                  margin: 2rem 1%;
                  flex: 1 1 30%;
                  justify-content: space-around;
                `}
                key={image.name}
              >
                <img
                  css={css`
                    max-width: 100%;
                    max-height: 3rem;
                  `}
                  src={image.src}
                  alt={`category img for ${image.name}`}
                />
                <span
                  css={css`
                    color: ${image.select ? colors.grey100 : colors.black};
                  `}
                >
                  {image.name}
                </span>
              </button>
            ))}
          </div>
        </>
      </Modal>
    </>
  );
}
