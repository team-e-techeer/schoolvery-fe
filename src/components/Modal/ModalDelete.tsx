import { css } from '@emotion/react';
import colors from '@/constants/colors';
import { useState } from 'react';
import styled from '@emotion/styled';
import Modal from './Modal';
import { useNavigate } from 'react-router-dom';
import Button from '../Button';

interface Props {
  visible: boolean;
  setModalVisible: React.Dispatch<React.SetStateAction<boolean>>;
}

interface Info {
  name: string;
  select: boolean;
}

const BtnDiv = styled.div`
  display: flex;
  width: 90%;
  margin: 2rem auto;
  border: 1px solid ${colors.grey300};
  border-radius: 8px;
`;

export default function ModalDelete({ visible, setModalVisible }: Props) {
  const navigate = useNavigate();
  const [checkDelete, setCheckDelete] = useState(false);
  const onClick = () => {
    setCheckDelete(true);
  };
  return (
    <>
      <Modal visible={visible} setModalVisible={setModalVisible}>
        <>
          <span
            css={css`
              display: flex;
              font-size: 2rem;
              margin: 1rem;
              margin-left: 10%;
            `}
          >
            글을 삭제하시겠습니까?
          </span>
          <BtnDiv>
            <Button height={5} buttonId="deleteA" style={{ marginRight: 5 }}>
              확인
            </Button>
            <Button height={5} buttonId="deleteB" style={{ marginLeft: 5 }} onClick={onClick}>
              취소
            </Button>
          </BtnDiv>
        </>
      </Modal>
    </>
  );
}
