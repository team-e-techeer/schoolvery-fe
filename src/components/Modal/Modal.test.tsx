import { fireEvent, render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Modal from './Modal';

describe('[Modal Component]', () => {
  it('[Modal Component] 렌더링 테스트', () => {
    const handleClose = jest.fn();
    const { container } = render(
      <Modal visible={true} setModalVisible={handleClose}>
        <div>hi</div>
      </Modal>,
      { wrapper: BrowserRouter }
    );
    expect(screen.getByText('hi')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
  it('[Modal Component] 닫기 버튼 누르기', async () => {
    const handleClose = jest.fn();
    const { container, getByText } = render(
      <Modal visible={true} setModalVisible={handleClose}>
        <div>hi</div>
      </Modal>,
      { wrapper: BrowserRouter }
    );
    expect(getByText('hi')).toBeTruthy();
    const button = screen.getByTestId('test-modal-button');
    fireEvent.click(button);

    expect(handleClose).toHaveBeenCalledTimes(1);
    expect(container).toMatchSnapshot();
  });
});
