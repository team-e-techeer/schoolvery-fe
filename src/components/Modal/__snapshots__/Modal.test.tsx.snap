// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Modal Component] [Modal Component] 닫기 버튼 누르기 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-box-pack: right;
  -ms-flex-pack: right;
  -webkit-justify-content: right;
  justify-content: right;
}

<div>
  <div
    class="one out"
    id="modal-container"
  >
    <div
      class="modal-background"
    >
      <div
        class="modal"
      >
        <button
          class="emotion-0"
          data-testid="test-modal-button"
        >
          <svg
            fill="currentColor"
            height="25"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 1024 1024"
            width="25"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 0 0 203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
            />
          </svg>
        </button>
        <div>
          hi
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`[Modal Component] [Modal Component] 렌더링 테스트 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-box-pack: right;
  -ms-flex-pack: right;
  -webkit-justify-content: right;
  justify-content: right;
}

<div>
  <div
    class="one"
    id="modal-container"
  >
    <div
      class="modal-background"
    >
      <div
        class="modal"
      >
        <button
          class="emotion-0"
          data-testid="test-modal-button"
        >
          <svg
            fill="currentColor"
            height="25"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 1024 1024"
            width="25"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 0 0 203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
            />
          </svg>
        </button>
        <div>
          hi
        </div>
      </div>
    </div>
  </div>
</div>
`;
